python/cudf/cudf/core/index.py

```diff
@@ -1929,7 +1929,40 @@ def floor(self, field):
         DatetimeIndex(['2020-05-31 08:59:00', '1999-12-31 18:44:00'],
         dtype='datetime64[ns]', freq=None)
         """
-        out_column = self._values.floor(field)
+        out_column = self._values.floor(freq)
+
+        return self.__class__._from_data({self.name: out_column})
+
+    def round(self, freq):
+        """
+        Perform round operation on the data to the specified freq.
+
+        Parameters
+        ----------
+        freq : str
+            One of ["D", "H", "T", "min", "S", "L", "ms", "U", "us", "N"].
+            Must be a fixed frequency like 'S' (second) not 'ME' (month end).
+            See `frequency aliases <https://pandas.pydata.org/docs/\
+                user_guide/timeseries.html#timeseries-offset-aliases>`__
+            for more details on these aliases.
+
+        Returns
+        -------
+        DatetimeIndex
+            Index of the same type for a DatetimeIndex
```

Comment:
The pandas docs say:

> Index of the same type for a DatetimeIndex or TimedeltaIndex, or a Series with the same index for a Series.

In the pandas code, this is indicating that the return type is the same as the _input_ type. Is this function used in cuDF for `DatetimeIndex`, `TimedeltaIndex`, and `Series`? If this is just for `DatetimeIndex` then we can try something like a short description (suggested below). If multiple input types are possible, then I would fix these docstrings to align with the pandas docs (to indicate that the return type varies with the input type).

```suggestion
        DatetimeIndex
            Index containing rounded datetimes.
```

Reply:
This is a good point @bdice ! In this case, `pandas` uses the same documentation across `dt`, `DatetimeIndex`, and `TimedeltaIndex` for `ceil`, `floor` and `round`. This PR adds `round` for `DatetimeIndex` and a datetime `Series`.

Your suggestion is actually better here because we have specific docs for each method and this method is just for `DatetimeIndex` as you pointed out, I will make that change. Thanks!